File: RepoTransBench/target_projects/Python/C++/ai-forever_gigachat/public_tests/test_public_client.cpp
#include <gtest/gtest.h>
#include <string>
#include <map>

std::map<std::string, std::string> _unknown_kwargs(std::map<std::string, std::string> kwargs) {
    // Identity for this dummy test, just returns its input
    return kwargs;
}

TEST(PublicClientTest, UnknownKwargsGetsFiltered) {
    std::map<std::string, std::string> input = {{"alpha", "beta"}, {"gamma", "42"}, {"is_test", "1"}};
    std::map<std::string, std::string> result = _unknown_kwargs(input);
    ASSERT_EQ(result["alpha"], "beta");
    ASSERT_EQ(result["gamma"], "42");
    ASSERT_EQ(result["is_test"], "1");
}

File: RepoTransBench/target_projects/Python/Rust/ai-forever_gigachat/public_tests/test_public_client.rs
use std::collections::HashMap;

// Simulates the Python "_unknown_kwargs" test
fn _unknown_kwargs(kwargs: HashMap<&str, &str>) -> HashMap<&str, &str> {
    kwargs
}

#[test]
fn test_unknown_kwargs_gets_filtered() {
    let mut args = HashMap::new();
    args.insert("alpha", "beta");
    args.insert("gamma", "42");
    args.insert("is_test", "true");
    let result = _unknown_kwargs(args.clone());
    assert_eq!(result.get("alpha"), Some(&"beta"));
    assert_eq!(result.get("gamma"), Some(&"42"));
    assert_eq!(result.get("is_test"), Some(&"true"));
}